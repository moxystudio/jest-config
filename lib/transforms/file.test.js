'use strict';

const fs = require('fs');
const tmp = require('tmp-sync');
const inlineTransformer = require('./file');

const dir = tmp.in(__dirname);

afterAll(() => {
    tmp.clean();
});

it('should return basename of the file', () => {
    const contents = 'foo';
    const filename = `${dir}/inline-test-file.txt`;

    fs.writeFileSync(filename, contents);

    const output = inlineTransformer.process(contents, filename, {});

    expect(output).toContain('inline-test-file.txt');
    expect(output).toMatchSnapshot();
});
