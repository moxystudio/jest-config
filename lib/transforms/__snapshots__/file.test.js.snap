// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return basename of the file 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;
const basename = \\"inline-test-file.txt\\";
var _default = basename;
exports.default = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImlubGluZS10ZXN0LWZpbGUudHh0Il0sIm5hbWVzIjpbImJhc2VuYW1lIl0sIm1hcHBpbmdzIjoiOzs7Ozs7QUFDQSxNQUFNQSxRQUFRLEdBQUcsc0JBQWpCO2VBRWVBLFEiLCJzb3VyY2VzQ29udGVudCI6WyJcbmNvbnN0IGJhc2VuYW1lID0gXCJpbmxpbmUtdGVzdC1maWxlLnR4dFwiO1xuXG5leHBvcnQgZGVmYXVsdCBiYXNlbmFtZTtcbiJdfQ=="
`;
